app/src/main/java/com/example/tasky/itemdetail/presentation/viewmodel/EventDetailViewModel.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.tasky.itemdetail.presentation.viewmodel
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.tasky.TaskyRoutes
+import com.example.tasky.itemdetail.domain.EventRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class EventDetailViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val repository: EventRepository
+) : ViewModel() {
+
+    var state by mutableStateOf(EventDetailState())
+        private set
+
+    init {
+        val eventId = savedStateHandle.get<String>(TaskyRoutes.EventDetailScreen.EVENT_ID)
+        if (eventId != null) {
+            state = state.copy(eventId = eventId)
```

Comment:
Nitpick, but this could also be assigned right in the default state above, so you save one copy call